src/Differ/DiffColorizer.php

```diff
@@ -99,16 +99,17 @@ private function inlineDiff(string $previousLine, string $nextLine, string $left
         $previousLineLength = mb_strlen($previousLine);
         $nextLineLength = mb_strlen($nextLine);
 
-        $start = $previousLineLength;
-
-        while ($start !== 0 && mb_strpos($nextLine, mb_substr($previousLine, 0, $start)) !== 0) {
-            --$start;
+        for ($start = $previousLineLength; $start !== 0; --$start) {
+            if (mb_strpos($nextLine, mb_substr($previousLine, 0, $start)) === 0) {
+                break;
+            }
         }
 
-        $end = $start;
-
-        while ($end < $previousLineLength && mb_strrpos($nextLine, $t = mb_substr($previousLine, $end), $start) !== ($nextLineLength - mb_strlen($t))) {
-            ++$end;
+        for ($end = $start; $end < $previousLineLength; ++$end) {
+            $t = mb_substr($previousLine, $end);
```

Comment:
I would love to name this variable better, but atm I don't have a understanding of this class.

input welcome.